dev/tools/gen_keycodes/lib/macos_code_gen.dart

```diff
@@ -5,62 +5,121 @@
 import 'package:path/path.dart' as path;
 
 import 'base_code_gen.dart';
-import 'key_data.dart';
+import 'constants.dart';
+import 'logical_key_data.dart';
+import 'physical_key_data.dart';
 import 'utils.dart';
 
+const List<String> kModifiersOfInterest = <String>[
+  'ShiftLeft',
+  'ShiftRight',
+  'ControlLeft',
+  'ControlRight',
+  'AltLeft',
+  'AltRight',
+  'MetaLeft',
+  'MetaRight',
+];
+
+// The name of keys that require special attention.
+const List<String> kSpecialPhysicalKeys = <String>['CapsLock'];
```

Comment:
Why doesn't this include `NumLock`?

Reply:
Ahh, I thought NumLock also followed that same weird convention. That's super odd that it's only CapsLock, but OK.

Reply:
Well the fact is that I've not seen a macOS that supports NumLock. The macOS embedding doesn't even support other locks.

Reply:
True, Apple branded keyboards don't use num lock, but you can plug in a non-apple keyboard that has numlock, and it works, so I assumed there were similar events.